vcert/connection_cloud.py

```diff
@@ -607,22 +622,122 @@ def set_policy(self, zone, policy_spec):
+            if for_update and len(owners_list) > 0:
+                log.debug("No owners provided in the policy specification")
+            else:
+                owner_list = build_owner_json(owners_list)
+                app_details.owner_ids_and_types = owner_list
+            if app_details.cit_alias_id_map:
                 # Only link cit with Application when cit is not already associated with Application
-                app_req = build_app_update_request(app_details, resp_cit_data)
-                status, data = self._put(URLS.APP_BY_ID.format(app_details.app_id), app_req)
-                if status != HTTPStatus.OK:
-                    raise VenafiError(f"Could not update Application [{app_name}] with cit [{resp_cit_data}]")
+                cit_map = app_details.cit_alias_id_map
+                cit_id, cit_name = get_cit_data_from_response(cit_data)
+                cit_map[cit_name] = cit_id
+            app_req = build_app_update_request(app_details, cit_map)
+            status, data = self._put(URLS.APP_BY_ID.format(app_details.app_id), app_req)
+            if status != HTTPStatus.OK:
+                raise VenafiError(f"Could not update Application [{app_name}] with cit [{resp_cit_data}]")
         else:
             # Application does not exist. Create one
-            app_req = build_app_create_request(app_name, user_details, resp_cit_data)
+            app_req = build_app_create_request(app_name, owners_list, resp_cit_data)
             status, data = self._post(URLS.APPLICATIONS, app_req)
             if status != HTTPStatus.CREATED:
                 raise VenafiError(f"Could not create application [{app_name}]")
         return
 
+    def resolve_owners(self, users_list, user_details):
+        owners_list = list()
+        for_update = False
+        if not users_list:
+            # When no users are provided on the list, adds apikey user as owner
+            api_owner = resolve_apikey_owner(user_details)
+            owners_list.append(api_owner)
+            for_update = True
+        else:
+            # Resolving the usernames list
+            # Creating a higher level Teams object to cache the response
+            teams_list = list()
+            users_list = list(set(users_list))
+            for user in users_list:
+                cloud_owner = self.resolve_user_to_cloud_owner(user)
+                if cloud_owner:
+                    owners_list.append(cloud_owner)
+                else:
+                    if not teams_list:
+                        status, response = self._get(URLS.TEAMS_ID)
+                        data_team = response['teams']
+                        for t in data_team:
+                            team = build_team(t)
+                            teams_list.append(team)
+                    team_owner = self.resolve_user_to_cloud_team(teams_list, user)
+                    if not team_owner:
+                        raise VenafiError(f"Unable to find team [{user}]")
```

Comment:
Change this method to inlcude the "user" word in the error message

Reply:
To be a little bit more clear, this error is raised given it was not possible to find an entity user/team that matches with the user name iterated at this time.